x/evm/statedb/statedb.go

```diff
@@ -423,83 +438,154 @@ func (s *StateDB) Transfer(sender, recipient common.Address, amount *big.Int) {
 
 	coins := sdk.NewCoins(sdk.NewCoin(s.evmDenom, sdkmath.NewIntFromBigIntMut(amount)))
 	senderAddr := sdk.AccAddress(sender.Bytes())
+	senderBalance := s.GetBalance(sender)
 	recipientAddr := sdk.AccAddress(recipient.Bytes())
+	recipientBalance := s.GetBalance(recipient)
 	if err := s.ExecuteNativeAction(common.Address{}, nil, func(ctx sdk.Context) error {
 		return s.keeper.Transfer(ctx, senderAddr, recipientAddr, coins)
 	}); err != nil {
 		s.err = err
 	}
+
+	if s.err == nil {
```

Comment:
@arrivets Actually not sure about this one. Would we want to trace event if the native action has returned an error?